internal/updater/utils/html/parsing.go

```diff
@@ -0,0 +1,70 @@
+package utils
+
+import (
+	"errors"
+
+	"github.com/andybalholm/cascadia"
+	"golang.org/x/net/html"
+)
+
+var (
+	ErrAttrNotFound = errors.New("matching attribute not found")
+)
+
+func GetText(n *html.Node) string {
+	return n.FirstChild.Data
+}
+
+func GetAttr(n *html.Node, key string) (string, error) {
+	for _, attr := range n.Attr {
+		if attr.Key == key {
+			return attr.Val, nil
+		}
+	}
+
+	return "", ErrAttrNotFound
+}
+
+func CheckID(n *html.Node, id string) bool {
+	if n.Type == html.ElementNode {
+		s, err := GetAttr(n, "id")
+
+		if err == nil && s == id {
+			return true
+		}
+	}
+
+	return false
+}
+
+func Traverse(n *html.Node, id string) *html.Node {
```

Comment:
Please add a comment above `Traverse`, this was a bit obscure when I first saw it.
I guess you can mention is does a depth first search too.

Reply:
Renamed this to `GetNodeByID`, since that was basically a better-named wrapper for it anyway.